thijsvanloef/palworld-server-docker
# Updated isWritable

## Context 

* Addressing #358 

## Choices

* Since `[ -w path ]` won't work if a directory is writable but not delete-able we check if it is a directory and if so then try to create a temporary file.

## Test instructions

1. Start normally. If this works then isWritable works for files and directories that are writable.
2. Verify the function still works for files that are not writable: `chmod 444 palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini`
3. Verify the function still works for directories that are writable and delete-able: `chmod 444 palworld/Pal/Saved/Config/LinuxServer/`
4. Verify the function works for directories that are writable but not delete-able

## Checklist before requesting a review
